### ISSUE TITLE
Error copying and pasting contents from a sheet

### ISSUE THREAD

**Issue by jprivillaso** [wijmo/Wijmo-Spread#3]
I was trying to copy the contents of a sheet to another sheet inside the same widget.

    var spread = $("#widget").wijspread("spread");
    var sheetA = spread.getSheetFromName("SheetA");
    var sheetB = spread.getSheetFromName("SheetB");

I am concerned why the following code does not work

    sheetB.fromJSON(sheetA.toJSON());

And this works

    sheetB.fromJSON(JSON.parse(JSON.stringify(sheetA.toJSON())));

The issue is that, when I use the first version, all the color formattings are black and using the second version everything pastes correctly and with the correct color and style formattings.

**Comment by GrapeCityDan** [comment]
Yes, it works correctly in second case because JSON.stringify turns a Javascript object into JSON text and stores that JSON text in a string. And JSON.parse turns a string of JSON text into a javascript object which is lately pushed into 2nd sheet.

I hope this helps.

**Comment by jprivillaso** [comment]
I understood perfectly, but my point is, if the method fromJSON(Obj) expects an Object as a parameter and toJSON() generates an object, why can´t I use it directly? 
Why do the code have to pass trough the stringify and parse methods to work?

**Comment by GrapeCityDan** [comment]
I am passing this to our support team. They will be better able to help

**Comment by jebro** [comment]
Hello,

Yes, it works correctly in second case because JSON.stringify turns a Javascript object into JSON text and stores that JSON text in a string. And JSON.parse turns a string of JSON text into a javascript object which is lately pushed into 2nd sheet.

Hope it clarifies.

Thanks,

Reeva
-- 
GrapeCity Support and Sales



-----Original Message-----
From: "Grapecity Sales and Support" <grapecityteam@grapecity.com>
Reply-To: "Grapecity Sales and Support" <grapecityteam@grapecity.com>
Date: Wed, 27 Jan 2016 03:41:47 -0800
To: "Dan Beall" <dan.beall@grapecity.com>
Cc: "inbox.nc@grapecity.com" <inbox.nc@grapecity.com>
Subject: Re: (Case 194794) [Wijmo-Spread] Error copying and pasting contents from a sheet (#3)

>Hello,
>
>Yes, it works correctly in second case because JSON.stringify turns a Javascript object into JSON text and stores that JSON text in a string. And JSON.parse turns a string of JSON text into a javascript object which is lately pushed into 2nd sheet.
>
>Hope it clarifies.
>
>Thanks,
>
>Reeva

**Comment by jebro** [comment]
Hello,

Would it be possible to send your contact information (name and email)?  This is an old GitHub repository that we are trying to close.

Regards
-- 
GrapeCity Support and Sales



-----Original Message-----
From: "Grapecity Sales and Support" <grapecityteam@grapecity.com>
Reply-To: "Grapecity Sales and Support" <grapecityteam@grapecity.com>
Date: Wed, 27 Jan 2016 03:41:47 -0800
To: "Dan Beall" <dan.beall@grapecity.com>
Cc: "inbox.nc@grapecity.com" <inbox.nc@grapecity.com>
Subject: Re: (Case 194794) [Wijmo-Spread] Error copying and pasting contents from a sheet (#3)

>Hello,
>
>Yes, it works correctly in second case because JSON.stringify turns a Javascript object into JSON text and stores that JSON text in a string. And JSON.parse turns a string of JSON text into a javascript object which is lately pushed into 2nd sheet.
>
>Hope it clarifies.
>
>Thanks,
>
>Reeva

**Comment by jprivillaso** [comment]
Hello,

Name: Juan Pablo Rivillas
Email: jprivillaso@gmail.com

Thanks
